### ISSUE TITLE
[Proposal] Migrate GatherPress Events Data to Custom Table Structure

### ISSUE THREAD

**Issue by bhubbard** [GatherPress/gatherpress#523]
### Is your enhancement related to a problem? Please describe.

Currently, the GatherPress plugin relies on custom post types within the WordPress ecosystem to store and manage data related to gatherings and events. While this approach aligns with WordPress conventions, it may not be the most efficient solution for handling the unique requirements and scale of the plugin as sites with large event totals continue to grow.

A notable example of the benefits of transitioning to a custom table structure can be found in WooCommerce's migration from custom post types to custom database tables for order management. This strategic shift resulted in significant performance gains, enabling WooCommerce to handle large volumes of orders more efficiently and effectively.

### Designs

No designs, as the new custom table should still use the gutenberg editor, blocks, etc.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by patriciabt** [comment]
From Feb 23, 2024 meeting notes:

> Going into “Central” application on dot ORG
> Birgit: Custom tables not acceptable by community leadership.
> Mike: It is unavoidable for a project like this.

For information and about another plugin, [this is the Slack conversation](https://wordpress.slack.com/archives/C02QB8GMM/p1707341024785899) about custom tables on the wp org network. 

[Otto's reply](https://wordpress.slack.com/archives/C02QB8GMM/p1707492443592639?thread_ts=1707341024.785899&cid=C02QB8GMM):

> I did look through it a bit, but I stopped when I saw that it uses custom tables.
> Using your own table will slow our progress down in getting it on org. Dotorg is a very big multi site system with several different database servers. Tables have to be specifically designed to work in it. Using your own tables in a plugin like this requires a lot more. Basically, we use hyperdb, and we will have to modify the settings for that to include the table. Additionally, the plugin can't create its own tables. We have to create it for them in advance.

Check with the meta team.

**Comment by mauteri** [comment]
Not sure I fully understand this change as GatherPress already leverages a custom table that it uses to join on the posts table. It does this for start and end datetimes as well as timezone which is where a bottleneck would happen when trying to query events in a datetime order.

**Comment by carstingaxion** [comment]
I guess this can be closed as answered. @mauteri what do you think?

**Comment by mauteri** [comment]
> I guess this can be closed as answered. [@mauteri](https://github.com/mauteri) what do you think?

Agreed on closing. I see the parallel to ecommerce stores hitting scale issues with millions of products, but events have a pretty different volume profile and abstracting them out of the posts table would mean giving up a lot of native WordPress conventions for a scaling problem that may never materialize. If we ever hit it in practice, this is worth reopening with the evidence.
